src/main/java/FrameworksDrivers/MainPageView.java

```diff
@@ -0,0 +1,145 @@
+package FrameworksDrivers;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import FrameworksDrivers.UIElements.Button;
+import FrameworksDrivers.UIElements.Icon;
+import InterfaceAdapters.MainPagePresenter;
+import InterfaceAdapters.OtherAccountPresenter;
+
+public class MainPageView implements ActionListener, View {
+    private JPanel masterPanel;
+    private Object[] paths;
+    private CardLayout layout;
+
+    private Button homeButton;
+    private Button chatButton;
+    private Button myProfileButton;
+    private Button likeButton;
+    private Button dislikeButton;
+    private Button viewOtherAccount;
+    
+    
+
+    public void createMainView(String iconFilePath){
+        //MainPage Frame
+        JFrame mainFrame = new JFrame();
+
+        // Create Lower Panel
+        JPanel lowerPanel = new JPanel();
+        lowerPanel.setLayout(new GridLayout(1, 3));
+
+        // Create HomeButton
+        homeButton = new Button();
+        homeButton.createButton(lowerPanel, "Home", 130,195,100,30);
+        homeButton.getButton().addActionListener(this);
+
+        // Create ChatButton
+        chatButton = new Button();
+        chatButton.createButton(lowerPanel, "Chats", 130,195,100,30);
+        chatButton.getButton().addActionListener(this);
+
+        // Create MyProfileButton
+        myProfileButton = new Button();
+        myProfileButton.createButton(lowerPanel, "My Profile", 130,195,100,30);
+        myProfileButton.getButton().addActionListener(this);
+
+        // Add these buttons to the lower panel
+        lowerPanel.add(homeButton.getButton());
+        lowerPanel.add(chatButton.getButton());
+        lowerPanel.add(myProfileButton.getButton());
+
+        // Create Middle Panel
+        JPanel middlePanel = new JPanel();
+        lowerPanel.setLayout(new GridLayout(1, 2));
+
+        // Create like button
+        likeButton = new Button();
+        likeButton.createButton(lowerPanel, "Like", 130,195,100,30);
+        likeButton.getButton().addActionListener(this);
+
+        // Create like button
+        dislikeButton = new Button();
+        dislikeButton.createButton(lowerPanel, "Dislike", 130,195,100,30);
+        dislikeButton.getButton().addActionListener(this);
+
+        // Add these buttons to the middle panel
+        middlePanel.add(likeButton.getButton());
+        middlePanel.add(dislikeButton.getButton());
+
+        // Create Top Panel
+        JPanel topPanel = new JPanel();
```

Comment:
Seems like the remaining JPanel (top/middle/lowerPanel) should be added as local attributes at the top.